controlplane/internal/controller/agentcontrolplane_controller.go

```diff
@@ -123,6 +128,17 @@ func (r *AgentControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		controllerutil.AddFinalizer(acp, acpFinalizer)
 	}
 
+	acpVersion, err := semver.NewVersion(acp.Spec.Version)
+	if err != nil {
+		log.Error(err, "invalid version", "version", acp.Spec.Version)
```

Comment:
```suggestion
		log.Error(err, "invalid version openshift", "version", acp.Spec.Version)
```
Minor: Maybe a good idea to let them know the version of what?